Switch to Tailwind and add API doc (#41)
netbirdio/docs: src/pages/docs/how-to/use-service-users-with-access-token.mdx

# Use Service Users with Access Token

## Service Users

Service users are a type of account that allow applications or services to access resources and perform actions within your organization.

> ### What are Service Users used for?

Reviewer: ```suggestion
### What can you use NetBird service users and access tokens for? 
```

> Service users are typically used to authenticate requests to a web service on behalf of an application or a service. They are an alternative to using a personal account, and can provide additional security by allowing you to limit the scope of access granted to the service user.

Reviewer: ```suggestion
```

> Service users are typically used to authenticate requests to a web service on behalf of an application or a service. They are an alternative to using a personal account, and can provide additional security by allowing you to limit the scope of access granted to the service user.

Reviewer: ```suggestion
```

> Service users can be used in a variety of scenarios, including:

Reviewer: ```suggestion
Most network management operations administrators do in the NetBird dashboard (https://app.netbird.io) can also be done via NetBird API using service users' access tokens.

The most common usage scenarios:
```

> - **Automated scripts**: If you need to run automated scripts that access resources within your organization, you can use service users instead of storing your login credentials in plain text.
> - **Third-party integrations**: If you want to integrate your resources with a third-party service, you can use a service user to provide access to your data without giving the third-party service access to your personal account.
> - **Access control**: If you need to grant access to a specific set of resources to a particular application or service, you can create a service user with the necessary permissions.

Reviewer: ```suggestion
- **Automated scripts**: if you need to run automated scripts or infrastructure as code tools like Ansible or Terraform that configure your NetBird network.
- **Third-party integrations**: if you want to integrate NetBird with a third-party service, you can use a service user to provide access to your data without giving the third-party service access to your personal account.
**Command-line tools**: if you use command-line tools like cUrl to interact with your NetBird network.
```

### Creating a Service User

> To create a service user, you'll need to log in to your organization's account and navigate to the "Service Users" section of your account settings.

Reviewer: ```suggestion
To create a service user, you'll need to log in to your organization's account at https://app.netbird.io and navigate to the "Users" -> "Service Users" section of your account.
```

> To create a service user, you'll need to log in to your organization's account and navigate to the "Service Users" section of your account settings.

Reviewer: ```suggestion
To create a service user, you'll need to log in to your organization's account at https://app.netbird.io and navigate to the "Users" -> "Service Users" section of your account.
```

Personal access tokens can be used in a variety of scenarios, including:

- **Automated scripts**: If you need to run automated scripts that access your account, you can use personal access tokens instead of storing your login credentials in plain text.
- **Third-party integrations**: If you want to integrate your account with a third-party service, you can use a personal access token to provide access to your data without giving the third-party service your login credentials.
- **Command-line tools**: If you use command-line tools to interact with your account, you can use personal access tokens to authenticate those requests.

### Creating a Personal Access Token

To create a personal access token, you'll need to log in to your account and navigate to the "Users" section and look for your user or create a [service user](#service-users) to use for your API requests.

<p align="center">
    <img src="/img/overview/personal-access-token-overview.png" alt="personal-access-token-overview" width="780" style={{boxShadow: '0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19)'}} />
</p>

> From there, you can create a new token and specify the permissions that the token should have.

Reviewer: ```suggestion
From there, you can create a new token and specify expiration for the token. You won't be able to modify your token.
```

<p align="center">
    <img src="/img/overview/personal-access-token-creation.png" alt="personal-access-creation-popup" width="400" style={{boxShadow: '0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19)'}} />
</p>

After the token was generated successfully you will see a plain version of your token to copy and store in a secure place.
Be aware that once you close the popup it is impossible to see the plain version of the token again as NetBird only stores a hashed version of the token.

<p align="center">
    <img src="/img/overview/personal-access-token-example.png" alt="personal-access-token-example" width="400" style={{boxShadow: '0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19)'}} />
</p>

<Note>
    It's important to keep your personal access tokens secure, as they can provide access to sensitive data and actions within your account. You should treat your personal access tokens like you would treat your password and never share them with anyone else.
</Note>

> ### Using Personal Access Tokens

Reviewer: ```suggestion
### Using access tokens
```
